core/trino-main/src/main/java/io/trino/sql/planner/SubqueryPlanner.java

```diff
@@ -132,25 +141,25 @@ public PlanBuilder handleSubqueries(PlanBuilder builder, io.trino.sql.tree.Expre
      * Find subqueries from the candidate set that are children of the given parent
      * and that have not already been handled in the subplan
      */
-    private <T extends io.trino.sql.tree.Expression> List<T> selectSubqueries(PlanBuilder subPlan, io.trino.sql.tree.Expression parent, List<T> candidates)
+    private <T extends io.trino.sql.tree.Expression> List<T> selectSubqueries(PlanBuilder subPlan, Iterable<Node> allSubExpressions, List<T> candidates)
+    {
+        return candidates
+                .stream()
+                .filter(candidate -> stream(allSubExpressions).anyMatch(child -> child == candidate))
+                .filter(candidate -> !subPlan.canTranslate(candidate))
+                .collect(toImmutableList());
+    }
+
+    private SuccessorsFunction<Node> recurseExpression(PlanBuilder subPlan)
```

Comment:
inline this?

Reply:
subPlan needs to be a final variable in order to be used in lambda - hence the method

Reply:
oh ok

Reply:
After extracting `allSubExpressions`, there is no more need for adding those conditions: `if (!subqueries.getSubqueries().isEmpty())`, or optimizing the `selectSubqueries()` method (as in the [comment](https://github.com/trinodb/trino/pull/25497#pullrequestreview-2739277967)).

Let's merge https://github.com/trinodb/trino/pull/25497 when ready, and then rebase your change, is that ok?

Reply:
Done.